validator/valnode/valnode.go

```diff
@@ -107,18 +113,28 @@ func CreateValidationNode(configFetcher ValidationConfigFetcher, stack *node.Nod
 	if err != nil {
 		return nil, err
 	}
-	var serverAPI *ExecServerAPI
-	var jitSpawner *server_jit.JitSpawner
+	var (
+		serverAPI    *ExecServerAPI
+		jitSpawner   *server_jit.JitSpawner
+		redisSpawner *arbredis.ExecutionSpawner
+	)
+	if config.RedisExecRunner.Enabled() {
+		es, err := arbredis.NewExecutionSpawner(&config.RedisExecRunner, arbSpawner)
+		if err != nil {
+			log.Error("creating redis execution spawner", "error", err)
+		}
+		redisSpawner = es
+	}
 	if config.UseJit {
 		jitConfigFetcher := func() *server_jit.JitSpawnerConfig { return &configFetcher().Jit }
 		var err error
 		jitSpawner, err = server_jit.NewJitSpawner(locator, jitConfigFetcher, fatalErrChan)
 		if err != nil {
 			return nil, err
 		}
-		serverAPI = NewExecutionServerAPI(jitSpawner, arbSpawner, arbConfigFetcher)
+		serverAPI = NewExecutionServerAPI(jitSpawner, arbSpawner, redisSpawner, arbConfigFetcher)
 	} else {
-		serverAPI = NewExecutionServerAPI(arbSpawner, arbSpawner, arbConfigFetcher)
+		serverAPI = NewExecutionServerAPI(arbSpawner, arbSpawner, redisSpawner, arbConfigFetcher)
```

Comment:
sereverAPI is a thin struct that implements rpc endpoints (see it's usage in valAPI).
It doesn't need and shouldn't be aware of the redisSpawner.
redisSpawner belongs directly in ValidationNode struct, like redisConsumer